File: Java/src/main/java/org/codecop/dependencies/extract_and_overide_call/Money.java
package org.codecop.dependencies.extract_and_overide_call;

import java.math.BigDecimal;

public class Money {

    private final BigDecimal value;

    public Money(int value) {
        this(new BigDecimal(value));
    }

    public Money(BigDecimal value) {
        this.value = value;
    }

    public Money add(Money other) {
        return new Money(value.add(other.value));
    }

    public Money percentage(int p) {
        return new Money(value.multiply(new BigDecimal(p)).divide(new BigDecimal(100)));
    }

    public String format() {
        return String.format("%.2f", value);
    }

    public BigDecimal asBigDecimal() {
        return value;
    }
}

File: CSharp/DependencyBreakingKatas/ExtractAndOverrideCall/Money.cs
namespace Org.Codecop.Dependencies.ExtractAndOverrideCall
{
    public class Money
    {
        private readonly decimal value;

        public Money(int value)
        {
            this.value = value;
        }

        public Money(decimal value)
        {
            this.value = value;
        }

        public Money Add(Money other)
        {
            return new Money(value + other.AsBigDecimal());
        }

        public Money Percentage(int p)
        {
            return new Money(value * p / 100);
        }

        public string Format()
        {
            return string.Format("{0:0.00}", value);
        }

        public decimal AsBigDecimal()
        {
            return value;
        }
    }
}
